store/smt/proof_test.go

```diff
@@ -0,0 +1,68 @@
+package smt_test
+
+import (
+	"crypto/sha256"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	smtstore "github.com/cosmos/cosmos-sdk/store/smt"
+	"github.com/lazyledger/smt"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func TestProofOpInterface(t *testing.T) {
+	hasher := sha256.New()
+	tree := smt.NewSparseMerkleTree(dbm.NewMemDB(), hasher)
+	key := []byte("foo")
+	value := []byte("bar")
+	root, err := tree.Update(key, value)
+	require.NoError(t, err)
+	require.NotEmpty(t, root)
+
+	proof, err := tree.Prove(key)
+	require.True(t, smt.VerifyProof(proof, root, key, value, hasher))
+
+	storeProofOp := smtstore.NewProofOp(root, key, smtstore.SHA256, proof)
+	require.NotNil(t, storeProofOp)
+	// inclusion proof
+	r, err := storeProofOp.Run([][]byte{value})
+	assert.NoError(t, err)
+	assert.NotEmpty(t, r)
+	assert.Equal(t, root, r[0])
```

Comment:
instead of adding `t` to every assert or require I usually shadow the package names:
```
assert := assert.New(t)
require := require.New(t)
// and then:
assert.Equal(root, r[0])  // no need to pass `t`
```

Reply:
Nice one!